OpenRA.Mods.Cnc/Traits/EnergyWall.cs

```diff
@@ -45,68 +45,78 @@ public void RulesetLoaded(Ruleset rules, ActorInfo ai)
 		}
 	}
 
-	public class EnergyWall : Building, IObservesVariables, ITick, ITemporaryBlocker
+	public class EnergyWall : IObservesVariables, ITick, ITemporaryBlocker, INotifyAddedToWorld, INotifyRemovedFromWorld
 	{
-		readonly EnergyWallInfo info;
+		public readonly EnergyWallInfo Info;
+		public readonly IEnumerable<CPos> Footprint;
+
+		readonly Building building;
 		IEnumerable<CPos> blockedPositions;
 
 		// Initial state is active to match Building adding the influence to the ActorMap
 		// This will be updated by ConditionsChanged at actor creation.
-		bool active = true;
+		public bool Active = true;
```

Comment:
Making this public isn't very nice - using the condition and `KillsActorAbove` trait I suggested earlier would avoid this.